src/main/java/seedu/address/logic/commands/SortCommand.java

```diff
@@ -0,0 +1,71 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Comparator;
+
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.logic.parser.SortCommandParser;
+import seedu.address.model.Model;
+import seedu.address.model.person.Person;
+
+/**
+ * Sorts persons in SocialBook according to the specified field.
+ */
+public class SortCommand extends Command {
+    public static final String COMMAND_WORD = "sort";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Sorts all persons in the address book according to the given parameter.\n"
+            + "Example: " + COMMAND_WORD;
+
+    public static final String MESSAGE_SUCCESS = "Sorted all persons by %s";
+
+    private final String parameter;
+
+    /**
+     * Creates a SortCommand to sort the person list using the given {@code parameter}
+     */
+    public SortCommand(String parameter) {
+        assert parameter != null : "Parameter cannot be null";
+        this.parameter = parameter;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+
+        if (parameter.equals(SortCommandParser.NAME)) {
```

Comment:
I think it would be cleaner to use switch case here

Reply:
Thanks for the suggestion! Have improved the code quality by following this.